src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AuthorizationRule/SetAzureEventhubAuthorizationRules.cs

```diff
@@ -23,84 +23,47 @@ namespace Microsoft.Azure.Commands.EventHub.Commands
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
         [ResourceGroupCompleter]
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
-            Position = 1, ParameterSetName = NamespaceAuthoRuleParameterSet,
-            HelpMessage = "Namespace Name.")]
-        [Parameter(Mandatory = true, Position = 1, ParameterSetName = EventhubAuthoRuleParameterSet)]
+        [Parameter(Mandatory = true, ParameterSetName = NamespaceAuthoRuleParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = EventhubAuthoRuleParameterSet, ValueFromPipelineByPropertyName = true, Position = 1)]
         [ValidateNotNullOrEmpty]
         [Alias(AliasNamespaceName)]
         public string Namespace { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
-            Position = 2, ParameterSetName = EventhubAuthoRuleParameterSet,
-            HelpMessage = "EventHub Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = EventhubAuthoRuleParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "EventHub Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasEventHubName)]
         public string EventHub { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipelineByPropertyName = true,
-            Position = 3,
-            HelpMessage = "AuthorizationRule Name.")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = "AuthorizationRule Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasAuthorizationRuleName)]
         public string Name { get; set; }
 
-        [Parameter(Mandatory = true,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = AuthoRuleInputObjectParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
-        [Parameter(Mandatory = false,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = NamespaceAuthoRuleParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
-        [Parameter(Mandatory = false,
-            ValueFromPipeline = true,
-            Position = 4,
-            ParameterSetName = EventhubAuthoRuleParameterSet,
-            HelpMessage = "Relay AuthorizationRule Object.")]
+        [Parameter(Mandatory = true, ParameterSetName = AuthoRuleInputObjectParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "Relay AuthorizationRule Object")]
+        [Parameter(Mandatory = false, ParameterSetName = NamespaceAuthoRuleParameterSet, ValueFromPipeline = true, HelpMessage = "Relay AuthorizationRule Object")]
```

Comment:
even though there should not have ever been a position, you cannot remove it because it could break scripts.  Please go through the breaking changes csv and add back all positions that you deleted.  If you would like to deprecate this, you can add an obsolete message and remove it in the next breaking change.

Reply:
Reverted the position change for optional parameters